src/experimental/tools/download2go/index.ts

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AsyncSubject, ReplaySubject } from "rxjs";
+
+import { downloader } from "./apis/publicApi/download";
+import { constructOfflineManifest } from "./apis/publicApi/offlineDownload";
+import { setUpDb } from "./apis/transactionDB/dbSetUp";
+import {
+  checkForPauseAMovie,
+  checkForResumeAPausedMovie,
+  checkForSettingsAddMovie,
+  IndexDBError,
+} from "./utils";
+
+import EventEmitter from "../../../utils/event_emitter";
+import { IActiveSubs, IPauseSubject } from "./apis/dash/types";
+import {
+  IAddMovie,
+  IEventsEmitter,
+  IOptionsStarter,
+  IProgressBarBuilderAbstract,
+  IPublicAPI,
+  IStoredManifest,
+} from "./types";
+
```

Comment:
It would be great if you could, for each declared function in the D2G tool, write a little documentation in the jsdoc format, as we already do it in the RxPlayer 😄

Reply:
yea, I've done it for the rest of the function but not in the `IPublicAPI`, I will do it :)